server/base/src/test/java/org/apache/accumulo/server/conf/util/ZooInfoViewerTest.java

```diff
@@ -0,0 +1,437 @@
+      justification = "test output of generated output")
+  @Test
+  public void propTest() throws Exception {
+    String uuid = UUID.randomUUID().toString();
+    InstanceId iid = InstanceId.of(uuid);
+
+    ZooReader zooReader = createMock(ZooReader.class);
+    var instanceName = "test";
+    expect(zooReader.getChildren(eq(ZROOT + ZINSTANCES))).andReturn(List.of(instanceName))
+        .anyTimes();
+    expect(zooReader.getData(eq(ZROOT + ZINSTANCES + "/" + instanceName)))
+        .andReturn(uuid.getBytes(UTF_8)).anyTimes();
+
+    var sysPropBytes = propCodec
+        .toBytes(new VersionedProperties(123, Instant.now(), Map.of("s1", "sv1", "s2", "sv2")));
+    expect(zooReader.getData(eq(SystemPropKey.of(iid).getNodePath()), anyObject(Watcher.class),
+        anyObject(Stat.class))).andReturn(sysPropBytes).anyTimes();
+
+    var nsBasePath = ZooUtil.getRoot(iid) + ZNAMESPACES;
+    expect(zooReader.getChildren(nsBasePath)).andReturn(List.of("a")).anyTimes();
+    expect(zooReader.getData(eq(nsBasePath + "/a" + ZNAMESPACE_NAME)))
+        .andReturn("a_name".getBytes(UTF_8)).anyTimes();
+    var nsPropBytes =
+        propCodec.toBytes(new VersionedProperties(123, Instant.now(), Map.of("n1", "nv1")));
+    NamespaceId nsId = NamespaceId.of("a");
+    expect(zooReader.getData(eq(NamespacePropKey.of(iid, nsId).getNodePath()),
+        anyObject(Watcher.class), anyObject(Stat.class))).andReturn(nsPropBytes).anyTimes();
+
+    var tBasePath = ZooUtil.getRoot(iid) + ZTABLES;
+    expect(zooReader.getChildren(tBasePath)).andReturn(List.of("t")).anyTimes();
+    expect(zooReader.getData(eq(tBasePath + "/t" + ZTABLE_NAME)))
+        .andReturn("t_table".getBytes(UTF_8)).anyTimes();
+    var tPropBytes =
+        propCodec.toBytes(new VersionedProperties(123, Instant.now(), Map.of("t1", "tv1")));
+    TableId tid = TableId.of("t");
+    expect(zooReader.getData(eq(TablePropKey.of(iid, tid).getNodePath()), anyObject(Watcher.class),
+        anyObject(Stat.class))).andReturn(tPropBytes).anyTimes();
+    expect(zooReader.getData(tBasePath + "/t" + ZTABLE_NAMESPACE))
+        .andReturn("+default".getBytes(UTF_8)).anyTimes();
+
+    replay(zooReader);
+
+    NamespacePropKey nsKey = NamespacePropKey.of(iid, nsId);
+    log.trace("namespace base path: {}", nsKey.getBasePath());
+
+    String testFileName = "./target/zoo-info-viewer-" + System.currentTimeMillis() + ".txt";
+
+    ZooInfoViewer.Opts opts = new ZooInfoViewer.Opts();
+    opts.parseArgs(ZooInfoViewer.class.getName(),
+        new String[] {"--instanceId", uuid, "--print-props", "--outfile", testFileName});
+
+    ZooInfoViewer viewer = new ZooInfoViewer();
+    // InstanceId found = viewer.getInstanceId(zooReader, opts);
+    viewer.generateReport(InstanceId.of(uuid), opts, zooReader);
+    // assertEquals(InstanceId.of(uuid), found);
+
+    verify(zooReader);
+
+    String line;
+    try (BufferedReader in = new BufferedReader(new FileReader(testFileName, UTF_8))) {
```

Comment:
BufferedReader to read input like this is very old school style. You could use java.util.Scanner, or since the file is very small, just read all the lines with `Files.readAllLines(path)` and loop over that result.

Reply:
Used scanner